### PR TITLE
fix: support anyOf inside allOf

### PR DESCRIPTION
**Pull request by MoLow**
Fixes https://github.com/fastify/fastify/issues/4920

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -832,9 +833,19 @@ function buildValue (context, location, input) {
 
   let code = ''
 
-  if (type === undefined && (schema.anyOf || schema.oneOf)) {
+  if ((type === undefined || type === 'object') && (schema.anyOf || schema.oneOf)) {
     context.validatorSchemasIds.add(location.getSchemaId())
 
+    if (schema.type === 'object') {
+      context.wrapObjects = false
```
Comment (mcollina) [inline index.js:840]: This is set to false and never reset to true, why?

**Review by mcollina — COMMENTED** [review]
good work!

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
benchmarks looks ok at a first glance

**Comment by Uzlopak** [comment]
Anothet run with the new benchmarks.

**Comment by MoLow** [comment]
@mcollina @ivan-tymoshenko when is this expected to be released in Fastify itself?

**Pull request merged**